sbi/inference/posteriors/base_posterior.py

```diff
@@ -938,6 +938,9 @@ def _ensure_x_consistent_with_default_x(self, x: Tensor) -> None:
                 x.shape == self.default_x.shape
             ), f"""The shape of the passed `x` {x.shape} and must match the shape of `x`
             used during training, {self.default_x.shape}."""
+        # If there is no default_x we still have ensure the correct shape.
+        else:
+            process_x(x, self._x_shape, allow_iid_x=self._allow_iid_x)
```

Comment:
`_ = process_x(...)`

Reply:
we don't need the returned `x` because this function just checks and returns `None`